### PR TITLE
add dev for java images

### PR DESCRIPTION
**Pull request by strongjz**

### CODE REVIEW

File: images/jdk/configs/latest-dev.apko.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+environment:
+  LANG: en_US.UTF-8
+  JAVA_HOME: /usr/lib/jvm/openjdk
+
+paths:
+  - path: /home/build
+    type: directory
+    uid: 65532
+    gid: 65532
+    permissions: 0o755
```
Comment (jedsalazar) [inline images/jdk/configs/latest-dev.apko.yaml:35]: is using octal format the preferred way here?

File: images/jdk/configs/latest-dev.apko.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
+contents:
+  repositories:
+    - https://packages.wolfi.dev/os
+  keyring:
+    - https://packages.wolfi.dev/os/wolfi-signing.rsa.pub
+  packages:
+    - ca-certificates-bundle
```
Comment (imjasonh) [inline images/jdk/configs/latest-dev.apko.yaml:7]: Are we supposed to include `apk-tools` in `-dev` images?

File: images/jdk/configs/openjdk-11-dev.apko.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
+contents:
+  repositories:
+    - https://packages.wolfi.dev/os
+  keyring:
+    - https://packages.wolfi.dev/os/wolfi-signing.rsa.pub
+  packages:
+    - ca-certificates-bundle
```
Comment (imjasonh) [inline images/jdk/configs/openjdk-11-dev.apko.yaml:7]: ditto

**Review by jedsalazar — COMMENTED** [review]
lgtm - just one question

File: images/jdk/image.yaml [hunk 4]
```diff
@@ -10,4 +10,16 @@ versions:
       config: configs/openjdk-11.apko.yaml
       extractTagsFrom:
         package: openjdk-11
+        prefix: openjdk-
+  - apko:
+      config: configs/latest-dev.apko.yaml
+      extractTagsFrom:
+        package: openjdk-17-dev
```
Comment (imjasonh) [inline images/jdk/image.yaml:17]: There doesn't appear to be a Wolfi package named `openjdk-17-dev`, I think we want this to be `package: openjdk-17`, and `prefix: dev-`?

That will give us `:dev-openjdk-17-11.0.18-r1`, is that right?

Comment (strongjz) [reply images/jdk/image.yaml:17]: fixed

File: images/jdk/configs/latest-dev.apko.yaml (hunk 1 above)
Comment (strongjz) [reply images/jdk/configs/latest-dev.apko.yaml:36]: yes? https://github.com/chainguard-dev/apko/blob/main/examples/nginx.yaml#L32

File: images/jdk/image.yaml (hunk 4 above)
Comment (rawlingsj) [reply images/jdk/image.yaml:17]: I think @imjasonh comment is still valid, there isn't a package called `openjdk-17-dev` should this be `openjdk-17`?

File: images/jre/image.yaml [hunk 5]
```diff
@@ -3,13 +3,27 @@ versions:
       config: configs/latest.apko.yaml
       extractTagsFrom:
         package: openjdk-17
-        prefix: openjdk-
+        prefix: openjdk-jre
```
Comment (rawlingsj) [inline images/jre/image.yaml:6]: FWIW the `-jre` suffix was dropped in another PR two weeks ago https://github.com/chainguard-images/images/pull/256/files#diff-3313beffc0b154f1127c0db56645507d074031c287e20c280555f1d9e6e87d36L6.  I think that was driven by a view that it was redundant given `jre` appears in the image name.  @amouat @patflynn is that right?

Comment (patflynn) [reply images/jre/image.yaml:6]: yep that's correct.

File: images/jre/image.yaml [hunk 6]
```diff
@@ -3,13 +3,27 @@ versions:
...
       tags:
         - openjdk-jre-17 # TODO: deprecate this old tag
   - apko:
       config: configs/openjdk-11.apko.yaml
       extractTagsFrom:
         package: openjdk-11
-        prefix: openjdk-
+        prefix: openjdk-jre
```
Comment (rawlingsj) [inline images/jre/image.yaml:13]: Same as comment above `-jre` was dropped previously

**Review by rawlingsj — APPROVED** [review]

**Pull request merged**